cli/popper/gha.py

```diff
@@ -537,10 +559,114 @@ def singularity_start(self):
         return ecode
 
 
-class HostRunner(ActionRunner):
+class VagrantRunner(ActionRunner):
+    """Action runner to run Vagrant Actions.
     """
-    Host Action Runner Class.
+
+    def __init__(self, action, workspace, env, dry):
+        super(VagrantRunner, self).__init__(action, workspace, env, dry)
+
+    def run(self, reuse=False):
+        root = scm.get_git_root_folder()
+
+        if 'vagrant://' in self.action['uses']:
+            uses = self.action['uses'][10:]
+            box_name = uses
+            uses = uses.split('/')
+            box_url = 'https://app.vagrantup.com/' + \
+                uses[0] + '/boxes/' + uses[1]
+            path = os.path.abspath(os.path.join(
+                root, box_name
+            ))
+            self.vagrant = vagrant.Vagrant(path)
+            if not os.path.exists(path):
+                os.makedirs(path)
+            elif os.path.isfile(os.path.join(path, 'Vagrantfile')):
+                os.remove(os.path.join(path, 'Vagrantfile'))
+
+            self.vagrant_init(box_name, box_url)
+        elif './' in self.action['uses']:
+            path = os.path.abspath(os.path.join(
+                root, self.action['uses']
+            ))
+            self.vagrant = vagrant.Vagrant(path)
+        else:
+            path = os.path.abspath(os.path.join(self.action['repo_dir'],
+                                                self.action['action_dir']))
+            self.vagrant = vagrant.Vagrant(path)
+
+        if not reuse:
+            if self.vagrant_exists(path):
+                self.vagrant_destroy()
+
+        popper.cli.vagrant_list.append(self.vagrant)
+        e = self.vagrant_up()
+
+        if e != 0:
+            log.fail('Action {} failed!'.format(self.action['name']))
+
+        cmd = []
+        if self.action.get('runs', None):
+            cmd.extend(self.action['runs'])
+        if self.action.get('args', None):
+            cmd.extend(self.action['args'])
+        self.vagrant_ssh(' '.join(cmd))
+
+    def setenv(self, k, v):
```

Comment:
looks like this is not used, right? if it isn't, can we remove it?

Reply:
> Also, we need to deal with sharing $HOME and $GITHUB_WORKSPACE folders.

Can you please help me on how to pass these env vars inside the vagrant vm automatically? I was thinking of making a template Vagrantfile and populating it during `vagrant_init()`. Is there any other way round ?

Reply:
I do not think there is a workaround it. We will need to dynamically inject this into the Vagrantfile.

Reply:
Yeah, thanks